QuickOSM/core/parser/preset_parser.py

```diff
@@ -0,0 +1,90 @@
+""" parser of the default presets file """
+
+import xml.dom.minidom as xml
+
+from QuickOSM.qgis_plugin_tools.tools.resources import resources_path
+
+PRESET_PATH = resources_path('JOSM_preset', 'defaultpresets.xml')
+
+__copyright__ = 'Copyright 2021, 3Liz'
+__license__ = 'GPL version 3'
+__email__ = 'info@3liz.org'
+
+
+class PresetsParser:
+
+    def __init__(self):
+
+        self.file = xml.parse(PRESET_PATH)
+
+    def parser(self):
+
+        def fetch_keys(group, heirs_name: list):
+            ancestors_name.append(group.getAttribute('name'))
+
+            children = filter(lambda child: self.node_filter(child), group.childNodes)
+            for inner_group in children:
+                heirs_item_name = fetch_keys(inner_group, [])
+
+                name_completion = ancestors_name[0]
+                if len(ancestors_name) != 1:
+                    for ancestor in ancestors_name[1:]:
+                        name_completion += '/' + ancestor
+                ancestors_name.pop(-1)
+                data = {'type': 'group', 'ancestors': ancestors_name, 'heirs': heirs_item_name}
```

Comment:
Can you use `namedTuple` ?
It's easier to access attributes, readonly, for pythonic.